src/ui/controllers/profile.ts

```diff
@@ -128,9 +163,78 @@ export async function updateOrganisation(request: Request, response: Response) {
 	}
 }
 
+function displayTokenPage(request: Request, response: Response, errMsg: string, value: string, organisation: string) {
+	response.send(template.render(`profile/enterToken`, request, response, {
+		error: true,
+		msg: errMsg,
+		org: value,
+		organisation,
+		originalUrl: request.body.originalUrl,
+	}))
+}
+
+export async function checkTokenValidity(request: Request, response: Response) {
+
+	const value = request.body.org
+	let organisationalUnit
+
+	try {
+		const organisationResponse: any = await registry.getWithoutHal(value)
+		organisationalUnit = {
+			code: organisationResponse.data.code,
+			name: organisationResponse.data.name,
+			paymentMethods: organisationResponse.data.paymentMethods,
+		}
+
+	} catch (error) {
+		console.log(error)
+		throw new Error(error)
+	}
+
+	const code = organisationalUnit.code
+	const domain = request.user.userName.split("@")[1]
+	const token = request.body.token
+	const accessToken = request.user.accessToken
+
+	if (!token) {
+			displayTokenPage(request, response, "Please don't leave the token blank", value, organisationalUnit.name)
+	} else {
+		const checkTokenValidResponse: any = await registry.updateToken(code, domain, token, false, accessToken)
+		if (checkTokenValidResponse === "NONE") {
+			// call the insert function in CSRS to update tokini
+			setLocalProfile(request, 'department', organisationalUnit.code)
+			setLocalProfile(request, 'organisationalUnit', organisationalUnit)
+			try {
+				// make the check
+				await registry.patch('civilServants', {
+					organisationalUnit: request.body.org,
+				}, request.user.accessToken)
+			} catch (error) {
+				logger.error(error)
+				throw new Error(error)
+			}
+			request.session!.save(() =>
+					response.redirect((request.body.originalUrl) ? request.body.originalUrl : defaultRedirectUrl)
+				)
+			} else if (checkTokenValidResponse === "Token not found") {
```

Comment:
Use the http response code instead of a String. x 3

Reply:
Done